### ISSUE TITLE
The reproduction results using VIT backbone were poor.

### ISSUE THREAD

**Issue by wyb1018666** [zhuolingli/FewVS#6]
Hello author! I used this command to reproduce the experimental results of the VIT backbone: `python train.py --mode train_proj --backbone SMKD --dataset CIFAR_FS`. However, the evaluation results showed a significant discrepancy with the paper. Is it incorrect to use this command directly? What modifications are needed? Thank you.

**Comment by zhuolingli** [comment]
Hi! Thanks a lot for your interest in our work!
It would be great if you could try evaluating FewVS on other datasets as well. 
I will check the code and fix any issues asap.

**Comment by wyb1018666** [comment]
> Hi! Thanks a lot for your interest in our work! It would be great if you could try evaluating FewVS on other datasets as well. I will check the code and fix any issues asap.

Thank you for your reply! My reproduction results using VIT as the backbone on the CIFAR-FS and FC100 datasets were slightly worse than those in the paper. However, I successfully reproduced it on the miniImageNet dataset using VIT as the backbone, with results very close to those in the paper. Regarding the CIFAR-FS and FC100 datasets, is there a problem with the pre-trained model you uploaded, or is there a problem with your processing of these two datasets? This is just my personal guess, and I look forward to your revisions. This paper has been very inspiring to me, and I am eager to conduct research based on your work. Thank you!

**Comment by zhuolingli** [comment]
Hi!

By setting `--sem` to 0, I evaluated the baseline using only the ViT backbone (pure visual classification) on CIFAR-FS dataset, the result I obtained was around 76%. After applying FewVS, the performance improved to 81%. Although FewVS indeed brings a clear improvement, you are right, I mistakenly uploaded the 1-shot CIFAR-FS ViT pretrained weights.

Since the project is a bit old and I no longer have additional backups of those models, I recommend following the [official SMKD repo](https://github.com/HL-hanlin/SMKD) to obtain the correct ViT backbone for CIFAR-FS.

Thank you again for your careful reproduction and for pointing this out!

**Comment by wyb1018666** [comment]
> Hi!
> 
> By setting `--sem` to 0, I evaluated the baseline using only the ViT backbone (pure visual classification) on CIFAR-FS dataset, the result I obtained was around 76%. After applying FewVS, the performance improved to 81%. Although FewVS indeed brings a clear improvement, you are right, I mistakenly uploaded the 1-shot CIFAR-FS ViT pretrained weights.
> 
> Since the project is a bit old and I no longer have additional backups of those models, I recommend following the [official SMKD repo](https://github.com/HL-hanlin/SMKD) to obtain the correct ViT backbone for CIFAR-FS.
> 
> Thank you again for your careful reproduction and for pointing this out!

Thank you for your reply! As you mentioned, the accuracy you achieved using the VIT architecture is 81%, but the corresponding accuracy on the CIFAR dataset in the paper is 85.63%. I was able to reproduce it successfully on miniImageNet, so I don't think it's a problem with the VIT backbone. I suspect the reason might be the 1-shot and 5-shot weights in the weights folder you uploaded. When you say you didn't back them up, are you referring to these two files? Can they be obtained from the official SMKD library?

**Comment by zhuolingli** [comment]
Hi, all ViT backbone weights used in our paper are taken from the official SMKD repository. You can directly follow their instructions to perform the two-stage training and obtain the weights. Hope this helps~

**Comment by wyb1018666** [comment]
> Hi, all ViT backbone weights used in our paper are taken from the official SMKD repository. You can directly follow their instructions to perform the two-stage training and obtain the weights. Hope this helps~

Okay, I will try to use the SMKD method to obtain the weights, and then fine-tune it using your framework to see if I can achieve the accuracy described in the paper. I would like to conduct my research based on your work. Thank you for your reply!

**Comment by zhuolingli** [comment]
No problem. 
If there are no further questions, I will close this issue :-).
